File: app/src/main/java/com/anokmik/tripassistant/databinding/list/ViewPagerListChangedCallback.java
package com.anokmik.tripassistant.databinding.list;

import android.databinding.ObservableList;
import android.support.v4.view.PagerAdapter;

public final class ViewPagerListChangedCallback extends ObservableList.OnListChangedCallback {

    private final PagerAdapter adapter;

    public ViewPagerListChangedCallback(PagerAdapter adapter) {
        this.adapter = adapter;
    }

    @Override
    public void onChanged(ObservableList sender) {
        adapter.notifyDataSetChanged();
    }

    @Override
    public void onItemRangeChanged(ObservableList sender, int positionStart, int itemCount) {
        adapter.notifyDataSetChanged();
    }

    @Override
    public void onItemRangeInserted(ObservableList sender, int positionStart, int itemCount) {
        adapter.notifyDataSetChanged();
    }

    @Override
    public void onItemRangeMoved(ObservableList sender, int fromPosition, int toPosition, int itemCount) {
        adapter.notifyDataSetChanged();
    }

    @Override
    public void onItemRangeRemoved(ObservableList sender, int positionStart, int itemCount) {
        adapter.notifyDataSetChanged();
    }

}

File: app/src/main/kotlin/com/anokmik/tripassistant/databinding/list/ViewPagerListChangedCallback.kt
package com.anokmik.tripassistant.databinding.list

import android.databinding.ObservableList
import android.support.v4.view.PagerAdapter

class ViewPagerListChangedCallback<T>(private val adapter: PagerAdapter) : ObservableList.OnListChangedCallback<ObservableList<T>>() {

    override fun onChanged(sender: ObservableList<T>?) = adapter.notifyDataSetChanged()

    override fun onItemRangeChanged(sender: ObservableList<T>?, positionStart: Int, itemCount: Int) = adapter.notifyDataSetChanged()

    override fun onItemRangeInserted(sender: ObservableList<T>?, positionStart: Int, itemCount: Int) = adapter.notifyDataSetChanged()

    override fun onItemRangeMoved(sender: ObservableList<T>?, fromPosition: Int, toPosition: Int, itemCount: Int) = adapter.notifyDataSetChanged()

    override fun onItemRangeRemoved(sender: ObservableList<T>?, positionStart: Int, itemCount: Int) = adapter.notifyDataSetChanged()

}